### DISCUSSION TITLE
View models in collection

### DISCUSSION THREAD

**Discussion by Polfo** [spatie/ray#116] [Ideas]
It is nice to be able to fluently tap `->ray()` in a collection using the `ray()` collection macro.
It would be nice to expand this to models.

Currently, when a collection contains models, using `->ray()` often does not help because it is not possible to inspect deeply nested elements like `#attributes: array:7 [ …7]`
Would it be possible to use `model()` (or `ray_model()` to prevent name conflict) in the same way as `ray()` on a collection?

**Comment by freekmurze** [comment]
I'm not sure what you mean exactly, but I'm open for a PR around this.

If the implementation is clean and the PR contains tests, I'll consider merging it in.

**Reply by Polfo** [reply]
This discussion should be in spatie/laravel-ray instead of spatie/ray.

I created spatie/laravel-ray#85 to clarify what I mean in a first step.
The second step would be to have a `Collection::macro('models', ...`
